### ISSUE TITLE
Pylint does not identify errors about private objects access

### ISSUE THREAD

**Issue by vyahello** [PyCQA/pylint#3272]
Hey guys, I'm wondering why `pylint` doesn't drop any errors about access to private functions/classes/modules. Please see code snippet below:

```python
# b.py
import _c
from a import _A, _do_some


dir(_c)
dir(_A())
dir(_do_some())

```

### Steps to reproduce
run `pylint b.py` command

### Current behavior
pylint report shows `10.00/10` rate with no errors

### Expected behavior
It should drop some errors about accessing private objects (at least `W0212 protected-access` warning)

### pylint --version output
pylint 2.4.3
astroid 2.3.2
Python 3.6.5 (default, Aug 31 2019, 14:52:34)
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.10.44.4)]

**Comment by PCManticore** [comment]
Thanks for the report! It definitely should emit `protected-access`

**Comment by vyahello** [comment]
Hey, do we have some updates regarding this one?

**Comment by Pierre-Sassoulas** [comment]
Hey @vyahello, it seems like no one worked on it yet, do you want to contribute for this feature ?

**Comment by Pierre-Sassoulas** [comment]
Closing as ``import-private-name`` was introduced in https://github.com/PyCQA/pylint/pull/5610 by @areveny .

Current result if you activate the check:
```
b.py:2:0: C2701: Imported private module (_c) (import-private-name)
b.py:3:0: C2701: Imported private objects (_A, _do_some) (import-private-name)
```
